network.c

```diff
@@ -380,7 +433,8 @@ static ssize_t network_read_attr(const struct iio_attr *attr,
 	const struct iio_context *ctx = iio_device_get_context(dev);
 	struct iio_context_pdata *pdata = iio_context_get_pdata(ctx);
 
-	return iiod_client_attr_read(pdata->iiod_client, attr, dst, len);
+	return (ssize_t)translate_error_for_api(
+		(int)iiod_client_attr_read(pdata->iiod_client, attr, dst, len));
```

Comment:
I'm not really excited about the above and how easy it might be to forget about it. Not sure it works but here it goes an idea... Being this windows specific, what about trying to put/hide this inside network-windows.c? For instance in:

```
int network_get_error(void)
{
	return -WSAGetLastError();
}
```
We could already take care of translation and so we would make things consistent and more cleaner in the core network file. Ideally, in that file we should not have to care about being on windows or unix. We could the same  in functions like `set_blocking_mode()`, `do_create_socket()` and `do_create_socket()`.

And if the above works, I think we could have a generic (with posix handling) for:

`network_is_interrupted()` and `network_connect_in_progress()`

Thoughts?

Reply:
It would make more sense to have the translation inside the Windows specific files (network-windows.c and dns_sd_windows.c). I will give it another go.